### PR TITLE
[Azure] Add A Renderer For Azure DevOps Pipeline

### PR DESCRIPTION
**Pull request by weisunOW**
### Description

This PR adds a renderer to annotate build logs for Azure DevOps pipelines. The new `AzureDevOpsPipelinesRenderer` shares most formatting logic with `GitHubActionsRenderer`.

E.g.
```Shell
xcodebuild [Flags] 2>&1 | xcbeautify --renderer azure-devops-pipelines
```
Or
```Shell
swift test 2>&1 | xcbeautify --renderer azure-devops-pipelines
```

## References

Azure DevOps - [Task Commands](https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#task-commands)

### CODE REVIEW

File: Sources/XcbeautifyLib/Renderers/Microsoft/MicrosoftOutputRendering.swift
Comment (weisunOW) [inline Sources/XcbeautifyLib/Renderers/Microsoft/MicrosoftOutputRendering.swift]: I changed the annotation for these two methods to `.error` as it reflects what's shown in the Xcode.

**Comment by weisunOW** [comment]
I will revert the test back to `XCTest` sometime tonight (UTC+11).

File: Sources/XcbeautifyLib/Renderers/Microsoft/MicrosoftOutputRendering.swift
Comment (cpisciotta) [reply Sources/XcbeautifyLib/Renderers/Microsoft/MicrosoftOutputRendering.swift]: Thanks! If it's not too much trouble, can you split this out into its own PR? Happy to do that if you'd like – just let me know if you'd like me to.

**Review by cpisciotta — COMMENTED** [review]
Hey @weisunOW! Thanks so much for your patience and contribution! This looks great to me, and I'm glad folks will be able to use this on Azure CI soon.

Quick question: Have you tested these changes in Azure's CI to see that it renders as you expect? If not, can you do that (and possibly link for me to check out)?

Thanks again... Looks good to approve once we address my one comment and question.

File: Sources/XcbeautifyLib/Renderers/Microsoft/MicrosoftOutputRendering.swift
Comment (weisunOW) [reply Sources/XcbeautifyLib/Renderers/Microsoft/MicrosoftOutputRendering.swift]: Sure, I will raise a separate PR soon.

Comment (weisunOW) [reply Sources/XcbeautifyLib/Renderers/Microsoft/MicrosoftOutputRendering.swift]: Done in #340, I will rebase this branch once #340 is merged.

**Comment by weisunOW** [comment]
Hi @cpisciotta, happy to help. 
> Quick question: Have you tested these changes in Azure's CI to see that it renders as you expect? If not, can you do that (and possibly link for me to check out)?

I've verified the branch on one of the pipelines of my enterprise project. The update renders fine on Azure Pipelines for the included logs under `Tests/ParsingTests`. However, when testing on my enterprise project,  it failed to annotate a few `PhaseScriptExecution` logs and some consecutive warning logs for both `GitHubActionsRenderer` and the new `AzureDevopsPipelineRenderer`. I couldn't figure out why because the Regex from the `CaptureGroup` matches fine on these missed logs.
   
Microsoft temporarily disabled free grants for new organisations on Azure DevOps until the end of this year, so I can't provide a public pipeline for you to check. Unfortunately, I also can't share the logs from the enterprise pipeline due to policy.

**Review by cpisciotta — APPROVED** [review]
LGTM! Thanks again for contributing this change 🚀

**Comment by cpisciotta** [comment]
@weisunOW It's late my time, so I will follow up on releasing a new version – likely tomorrow night.

**Comment by weisunOW** [comment]
Thanks @cpisciotta, looking forward to the new release.

**Comment by cpisciotta** [comment]
@weisunOW This is now available via [2.16.0](https://github.com/cpisciotta/xcbeautify/releases/tag/2.16.0). Homebrew should pick it up shortly. Thanks again!

**Pull request merged**
